MobilityData/mobility-feed-api
# feat: added WIP flag to feeds

**Summary:**
This pull request introduces visibility restrictions for work-in-progress (WIP) feeds in `feeds_api_impl.py` and `search_api_impl.py`, based on user email domains. It includes a new function to determine if a user’s email is restricted, updating query filters accordingly to enforce feed visibility rules. Additionally, a new SQL script is included to update the database schema and refresh the materialized view to support these changes.

**Expected Behavior:**
Feeds marked with an internal status of "WIP" are only visible to users with email domains ending in `@mobilitydata.org`. Users with other domains will not see these WIP feeds.

**Testing Instructions:**
1. Set up a local database and run the populate script.
2. Update the internal status of selected feeds to "WIP" in the database.
3. Run the API locally.
4. Make requests to the `gtfs`, `gtfs_rt`, and `feeds` endpoints, as well as the search functionality, while setting the `x-goog-authenticated-user-email` header to emails ending with both `@mobilitydata.org` and non-`@mobilitydata.org` domains.
5. Verify that only users with `@mobilitydata.org` emails can view WIP feeds.

Testers are encouraged to follow these instructions but are also welcome to explore other test cases to ensure robust behavior outside the listed scenarios.

Please make sure these boxes are checked before submitting your pull request - thanks!
